### ISSUE TITLE
Support polymorphism

### ISSUE THREAD

**Issue by Tamriel** [asyncapi/asyncapi-react#380]
The following example is found in the chapter **Composition and Inheritance (Polymorphism)** of the [specification](https://www.asyncapi.com/docs/specifications/2.0.0-rc1/).

Current behavior in the generated html: If `Pet` is used in a message payload, only the attributes of the Pet class are shown.

My desired behavior: A dropdown is shown next to the attribute `petType`. It has the entries `Cat` and `Dog`. Depending on the selection, additional attributes are shown.
This is the case in the OpenApi html generator [ReDoc](https://redocly.github.io/redoc/#operation/addPet).

````
schemas:
  Pet:
    type: object
    discriminator: petType
    properties:
      name:
        type: string
      petType:
        type: string
    required:
    - name
    - petType
  Cat:  ## "Cat" will be used as the discriminator value
    description: A representation of a cat
    allOf:
    - $ref: '#/components/schemas/Pet'
    - type: object
      properties:
        huntingSkill:
          type: string
          description: The measured skill for hunting
          enum:
          - clueless
          - lazy
          - adventurous
          - aggressive
      required:
      - huntingSkill
  Dog:  ## "Dog" will be used as the discriminator value
    description: A representation of a dog
    allOf:
    - $ref: '#/components/schemas/Pet'
    - type: object
      properties:
        packSize:
          type: integer
          format: int32
          description: the size of the pack the dog is from
          minimum: 0
      required:
      - packSize

**Comment by derberg** [comment]
@Tamriel hey, can you check current behaviour, HTML template (used in playground) changed a lot. Please update the issue if it is still valid

**Comment by magicmatatjahu** [comment]
@derberg react-component still doesn't support the `discriminator` property.

**Comment by dominik3131** [comment]
Any updates on this topic?

**Comment by magicmatatjahu** [comment]
@dominik3131 Hi. The discriminator is the only thing we don't currently render, and that's because we don't know how to render it nicely. I don't know if this will be implemented in the 1.0.0 release, probably later, but contributions are always welcome :) Why is this so hard to render?

The discriminator only works for schemes of type object, but the problem is how to render it when someone defines a scheme like:

```yaml
type: 
  type: [object, string, array]
  oneOf: ...
  allOf:
    - .. base schema with discriminiatir
    - .. new properties
  items: ...
```

From a spec point of view this is still correct, and currently `next` branch renders this, however how do we render the discriminator in such cases?
